Xamarin.Forms.Platform.Android/AppCompat/CheckBoxRenderer.cs

```diff
@@ -0,0 +1,258 @@
+			return new SizeRequest(new Size(MeasuredWidth, MeasuredHeight), MinimumSize());
+		}
+
+		void IVisualElementRenderer.SetElement(VisualElement element)
+		{
+
+			if (element == null)
+			{
+				throw new ArgumentNullException(nameof(element));
+			}
+
+			if (!(element is CheckBox checkBox))
+			{
+				throw new ArgumentException("Element is not of type " + typeof(CheckBox), nameof(element));
+			}
+
+			CheckBox oldElement = CheckBox;
+			CheckBox = checkBox;
+
+			Performance.Start(out string reference);
+
+			if (oldElement != null)
+			{
+				oldElement.PropertyChanged -= OnElementPropertyChanged;
+			}
+
+			element.PropertyChanged += OnElementPropertyChanged;
+
+			if (_tracker == null)
+			{
+				_tracker = new VisualElementTracker(this);
+
+			}
+
+			if (_visualElementRenderer == null)
+			{
+				_visualElementRenderer = new VisualElementRenderer(this);
+			}
+
+			Performance.Stop(reference);
+			this.EnsureId();
+						
+			UpdateOnColor();
+
+			ElementChanged?.Invoke(this, new VisualElementChangedEventArgs(oldElement, CheckBox));
+			CheckBox?.SendViewInitialized(Control);
+		}
+
+		// CheckBox related
+		void OnElementPropertyChanged(object sender, PropertyChangedEventArgs e)
+		{
+			if (e.PropertyName == CheckBox.TintColorProperty.PropertyName)
+			{
+				UpdateOnColor();
+			}
+
+			ElementPropertyChanged?.Invoke(this, e);
+		}
+
+		void IOnCheckedChangeListener.OnCheckedChanged(CompoundButton buttonView, bool isChecked)
+		{
+			((IElementController)CheckBox).SetValueFromRenderer(CheckBox.IsCheckedProperty, isChecked);
+		}
+
+		void UpdateOnColor()
+		{		
+
+			if (CheckBox == null || Control == null)
+				return;
+
+
+			var mode = PorterDuff.Mode.SrcIn;
+
+			var stateChecked = global::Android.Resource.Attribute.StateChecked;
+			var stateEnabled = global::Android.Resource.Attribute.StateEnabled;
+			var statePressed = global::Android.Resource.Attribute.StatePressed;
+
+			//Need to find a way to get this color out of Android somehow.
+			var uncheckedDefault = AColor.Gray;
+			var disabledColor = AColor.LightGray;
+
+			var tintColor = CheckBox.TintColor == Color.Default ? Color.Accent.ToAndroid() : CheckBox.TintColor.ToAndroid();
+
+			var list = new ColorStateList(
+					new int[][] 
+					{
+						new int[] { -stateEnabled, stateChecked },
+						new int[] { stateEnabled, stateChecked },
+						new int[] { stateEnabled, -stateChecked },
+						new int[] { stateEnabled, statePressed },
+						new int[] { },
+					},
+					new int[]
+					{
+						disabledColor,
+						tintColor,
+						tintColor,
+						tintColor,
+						disabledColor,
+					});
+				
+
+			if (Forms.IsLollipopOrNewer)
+			{
+				Control.ButtonTintList = list;
+				Control.ButtonTintMode = mode;
+			}
+			else
+			{
+				CompoundButtonCompat.SetButtonTintList(Control, list);
+				CompoundButtonCompat.SetButtonTintMode(Control, mode);
+			}
+				
+		}
+
+
+		// general state related
+		void IOnFocusChangeListener.OnFocusChange(AView v, bool hasFocus)
```

Comment:
I think you'll need this in the constructor
```C#
OnFocusChangeListener = this;
```

in order for this to run, and then make sure to null it out in the dispose